tests/unit/stores/pacbioLibraryBatches.spec.js

```diff
@@ -0,0 +1,153 @@
+import useRootStore from '@/stores'
+import {
+  createPinia,
+  setActivePinia,
+  successfulResponse,
+  failedResponse,
+} from '@support/testHelper.js'
+import { usePacbioLibraryBatchesStore } from '@/stores/pacbioLibraryBatches.js'
+import { beforeEach, describe, expect, vi } from 'vitest'
+import PacbioLibraryBatchFactory from '@tests/factories/PacbioLibraryBatchFactory.js'
+import PacbioRequestsFactory from '@tests/factories/PacbioRequestsFactory.js'
+import PacbioTagSetFactory from '@tests/factories/PacbioTagSetFactory.js'
+import { usePacbioRootStore } from '@/stores/pacbioRoot.js'
+import fs from 'fs'
+import { store } from '@support/testHelper.js'
+
+const pacbioRequestsFactory = PacbioRequestsFactory()
+const pacbioTagSetFactory = PacbioTagSetFactory()
+const pacbioLibraryBatchFactory = PacbioLibraryBatchFactory(pacbioTagSetFactory.storeData.tags)
+
+describe('usePacbioLibraryBatchesStore', () => {
+  const apiGetter = store.getters.api.v2
+
+  let rootStore, pacbioLibraryBatchStore, pacbioRootStore
+  beforeEach(() => {
+    /*Creates a fresh pinia instance and make it active so it's automatically picked
+      up by any useStore() call without having to pass it to it for e.g `useStore(pinia)`*/
+    const pinia = createPinia()
+    setActivePinia(pinia)
+    rootStore = useRootStore()
+    pacbioLibraryBatchStore = usePacbioLibraryBatchesStore()
+    pacbioRootStore = usePacbioRootStore()
+  })
+
+  describe('librariesInBatch', () => {
+    it('returns libraries in batch', () => {
+      pacbioLibraryBatchStore.libraries = pacbioLibraryBatchFactory.storeData.libraries
+      pacbioLibraryBatchStore.tubes = pacbioLibraryBatchFactory.storeData.tubes
+      pacbioRootStore.tags = pacbioTagSetFactory.storeData.tags
+      expect(pacbioLibraryBatchStore.librariesInBatch).toEqual(
+        pacbioLibraryBatchFactory.storeData.librariesInBatch,
+      )
+    })
+  })
+
+  describe('actions', () => {
+    describe('createLibraryBatch', () => {
+      let create, getTagSets, csvFile, csvFileTextContent, logMessage, mockSuccessResponse, tagSet
+
+      beforeEach(() => {
+        vi.spyOn(apiGetter.traction.pacbio.requests, 'get').mockResolvedValue(
```

Comment:
Do you need to spy on this? Can you do the same thing as below with the getTagSets?